src/compiler/checker.ts

```diff
@@ -8839,13 +8840,36 @@ namespace ts {
             })!.parent;
         }
 
+        function getInstanceTypeOfClassSymbol(classSymbol: Symbol): Type {
+            const classType = getDeclaredTypeOfSymbol(classSymbol) as GenericType;
+            if (!classType.typeParameters) {
+                return classType;
+            }
+            const variances = getVariances(classType);
+            const typeArguments = map(classType.typeParameters, (typeParameter, i) => {
+                if (!inferInstanceTypeArgumentsAsConstraint) {
+                    return anyType;
+                }
+                const variance = variances[i];
+                switch (variance) {

```

Comment:
You'll want to say ` & Variance.VarianceMask` here, to exclude the `Unmeasurable` and `Unreliable` markers from consideration, since they augment another variance kind.